File: 프로그래머스/lv2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    queue = deque([i, v] for i, v in enumerate(priorities))
    ans = 0
    
    while True:
        temp = queue.popleft()
        # 큐 안에 우선순위가 더 높은게 존재하는 경우가 하나라도 있으면 true가 됨
        if any(temp[1] < q[1] for q in queue):
            queue.append(temp)
        else:
            ans += 1
            if temp[0] == location:
                return ans

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
  let q = priorities.map((val, idx) => [idx, val]);
  let ans = 0;

  while (q.length) {
    let [idx, priority] = q.shift();

    // 큐에서 우선순위가 더 높은 프로세스가 있는지 확인
    if (q.some((item) => item[1] > priority)) {
      q.push([idx, priority]);
    } else {
      ans++;
      if (idx === location) {
        return ans;
      }
    }
  }
}
